.github/workflows/ci.yaml

```diff
@@ -27,21 +32,62 @@ jobs:
           CACHE_NUMBER: 0
         with:
           path: ~/conda_pkgs_dir
-          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/environment-py${{ matrix.python-version }}.yml') }}
+          key: ${{ runner.os }}-conda-${{ env.CACHE_NUMBER }}-${{ hashFiles('ci/environment.yml') }}
       - uses: conda-incubator/setup-miniconda@v2
         with:
           activate-environment: test_env_xmovie # Defined in ci/environment*.yml
           auto-update-conda: false
           python-version: ${{ matrix.python-version }}
-          environment-file: ci/environment-py${{ matrix.python-version }}.yml
+          environment-file: ci/environment.yml
           use-only-tar-bz2: true # IMPORTANT: This needs to be set for caching to work properly!
       - name: Set up conda environment
-        shell: bash -l {0}
         run: |
-          python -m pip install -e .
+          python -m pip install -e . --no-deps
           conda list
       - name: Run Tests
+        run: |
+          pytest --cov=./ --cov-report=xml
+      - name: Upload code coverage to Codecov
+        uses: codecov/codecov-action@v1
+        with:
+          file: ./coverage.xml
+          flags: unittests
+          env_vars: OS,PYTHON
+          name: codecov-umbrella
+          fail_ci_if_error: false
+
+  core-deps-dev:
```

Comment:
Somehow this is never even started? What am I doing wrong here. I pretty much copied this from xgcm